rfcs/OperationExpressions.md

```diff
@@ -0,0 +1,219 @@
+- `User.latestMedia` field (returns a `Media` union)
+- `Post` type in Media union
+- `title` field
+
+you might use a query string such as:
+
+```
+?docs=User.friends>latestMedia>Post.title
+```
+
+### Linking from a field description to an operation path
+
+If, for example, you were to deprecate a root-level field in your schema, you
+might want to indicate where the user can retrieve the equivalent data now. You
+could do this by including an operation expression as part of the deprecation
+reason:
+
+> The `Query.branchesFromFork` field is being removed; please use the following
+> path instead: `Query>repositories>forks>branches`
+
+### Indicating how to access a particular field
+
+When reading the documentation of a type in GraphiQL it currently does not
+indicate how to reach a particular field. Though there are often infinitely
+many paths to reach a field, often the shortest are the most valuable, so
+GraphiQL could indicate a few of the shorter paths using operation expression
+syntax:
+
+> `User.firstName` can be accessed through paths such as:
+>
+> - `>me>firstName`
+> - `>articles>author>firstName`
+> - `>searchMedia>Book.author>firstName`
+> - `mutation>createUser>user>firstName`
+
+### Analytics
+
+When analysing how a GraphQL schema is used, it may be useful to track
+statistics for each type, field, argument using Schema Coordinates; but it may
+also be interesting to track through what paths users are finding said fields.
+You could use operation expression syntax to track this:
+
+```
+counters['Query.cities>libraries>findBook(isbn:)']++
+```
+
+## Syntax
+
+Syntax is in flux; but here's some thoughts:
+
+#### Pathing
+
+Following a path from one field to the next could use the `>` character; this
+is already used in Apollo's GraphQL documentation browser and is intuitive for
+navigation. This leaves `.` available and non-ambiguous for referring to fields
+on a type, which is useful when disambiguating references on a union type, for
+instance:
+
+```
+>me>media>Film.duration
+```
+
+might model:
+
+```graphql
+{ me { media { ... on Film { duration } } } }
+```
+
+#### Operations
+
+The expression `>me>name` would expand to `{ me { name } }`.
+
+If you want to create a mutation or subscription operation, you can prefix the
+path with the operation type (you can do this for queries too, but just like in
+operation documents, the query keyword is optional):
+
+- `mutation>createUser>user>name` expands to `mutation ($input: CreateUserInput!) { createUser(input: $input) { user { name } } }`
+- `subscription>currentUserUpdated>name` expands to `subscription { currentUserUpdated { name } }`
+- `query>me>name` expands to `query { me { name } }`
+
+You may name operations by prefixing with an operation name followed by a
+colon; for example `MyName:>me>name` expands to `query MyName { me { name } }`.
+
+#### Fragments
+
+Fragments start with a type name followed by a period: `User.friends>name`
```

Comment:
How do you distinguish this from named operations? See https://github.com/graphql/graphql-spec/pull/823/files#r577833203

Reply:
Fragments start with a typename period fieldname, operations start with operation type (optionally) greater than, fieldname

fragment: `User.field` Or `Name:User.field`

operation: `query>field` `>field` `Name:mutation>field`